packages/turbo/ios/RNSession.swift

```diff
@@ -61,102 +61,86 @@ class RNSession: NSObject {
   }()
   public lazy var webView: WKWebView = turboSession.webView
   
-  func registerVisitableView(newView: RNSessionSubscriber) {
-    if (!visitableViews.contains {
-      $0.id == newView.id
-    }) {
-      visitableViews.append(newView)
+  func visitableViewDidDisappear(view: RNVisitableView) {
+    if (view === visitableView) {
+      self.visitableView = nil
     }
   }
   
-  func unregisterVisitableView(view: RNSessionSubscriber) {
-    let wasTopMostView = visitableViews.last?.id == view.id
-    
-    let viewIdx = visitableViews.lastIndex(where: {
-      view.id == $0.id
-    })
-    visitableViews.remove(at: viewIdx!)
-
-    // The new top-most view is not registered when the previous top-most view is a modal
-    if (wasTopMostView) {
-      guard let newView = visitableViews.last else {
-        return
-      }
-      visitableViewWillAppear(view: newView)
-    }
-  }
```

Comment:
All of this was to handle a modal being shown on top of a stack. Turbo Native also doesn't have support for this, and it caused quite a few inconsistencies and bugs, so keeping it simple and removing that. If a modal is to be shown, then we should not re-use the session.